src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -278,51 +279,70 @@ public void write(List<Object> values) {
                 + ") : "
                 + values);
       }
-
       recordConsumer.startMessage();
+      write(values, schema);
+      recordConsumer.endMessage();
+    }
+
+    private void write(List<Object> values, GroupType type) {
+      List<Type> cols = type.getFields();
       for (int i = 0; i < cols.size(); ++i) {
         Object val = values.get(i);
-        // val.length() == 0 indicates a NULL value.
         if (val != null) {
-          String fieldName = cols.get(i).getPath()[0];
+          String fieldName = cols.get(i).getName();
```

Comment:
while this may be functionally equivalent, I'd not mix this into your current PR. Feel free to send a separate PR to replace getPath[0] with getName.

Reply:
The `.getPath()[0]` was used against `ColumnDescriptor`. The `.getName()` is used on `Parquet.Type` as we need the information in `Parquet.Type`.